solver/llbsolver/solver.go

```diff
@@ -193,138 +197,186 @@ func (s *Solver) Solve(ctx context.Context, id string, sessionID string, req fro
 	}
 
 	var exporterResponse map[string]string
+	cacheExporters := exp.CacheExporters
 	if e := exp.Exporter; e != nil {
-		cached, err := result.ConvertResult(res, func(res solver.ResultProxy) (solver.CachedResult, error) {
-			return res.Result(ctx)
-		})
+		var inlineCacheExporter *RemoteCacheExporter
+		cacheExporters, inlineCacheExporter = splitCacheExporters(cacheExporters)
```

Comment:
Should this split be unconditional? If we reach a scenario where `exp.Exporter == nil`, we shouldn't attempt to build inline cache using `runCacheExporters` below, right?

Reply:
Ugh, right. Good point.